tree/dataframe/src/RLoopManager.cxx

```diff
@@ -387,40 +387,14 @@ RLoopManager::RLoopManager(ROOT::RDF::Experimental::RDatasetSpec &&spec)
 
    SetTree(std::move(chain));
 
-   // Create the friends from the list of friends
+   // Create friends from the specification and connect them to the main chain
    const auto &friendInfo = spec.GetFriendInfo();
-   const auto &friendNames = friendInfo.fFriendNames;
-   const auto &friendFileNames = friendInfo.fFriendFileNames;
-   const auto &friendChainSubNames = friendInfo.fFriendChainSubNames;
-   const auto nFriends = friendNames.size();
-
-   for (auto i = 0u; i < nFriends; ++i) {
-      const auto &thisFriendName = friendNames[i].first;
-      const auto &thisFriendAlias = friendNames[i].second;
-      const auto &thisFriendFiles = friendFileNames[i];
-      const auto &thisFriendChainSubNames = friendChainSubNames[i];
-
-      // Build a friend chain
-      auto frChain = std::make_unique<TChain>(thisFriendName.c_str());
-      const auto nFileNames = friendFileNames[i].size();
-      if (thisFriendChainSubNames.empty()) {
-         // If there are no chain subnames, the friend was a TTree. It's safe
-         // to add to the chain the filename directly.
-         for (auto j = 0u; j < nFileNames; ++j) {
-            frChain->Add(thisFriendFiles[j].c_str());
-         }
-      } else {
-         // Otherwise, the new friend chain needs to be built using the nomenclature
-         // "filename/treename" as argument to `TChain::Add`
-         for (auto j = 0u; j < nFileNames; ++j) {
-            frChain->Add((thisFriendFiles[j] + "/" + thisFriendChainSubNames[j]).c_str());
-         }
-      }
-
-      // Make it friends with the main chain
-      fTree->AddFriend(frChain.get(), thisFriendAlias.c_str());
-      // the friend trees must have same lifetime as fTree
-      fFriends.emplace_back(std::move(frChain));
+   fFriends = ROOT::Internal::TreeUtils::MakeFriends(friendInfo);
+   const auto nFriends = friendInfo.fFriendNames.size();
+   R__ASSERT(nFriends == fFriends.size() && "Created the wrong number of friends from the available information.");
```

Comment:
it feels like this check should be in `MakeFriends`, or in a `MakeFriends` unit test. `RLoopManager::RLoopManager` must be able to rely on the functions it calls to respect their contracts or error out themselves.

it should also probably be an `assert` rather than a `R__ASSERT`, we shouldn't test for bugs in production.

Reply:
Ok the check can be moved in `MakeFriends`

Reply:
Actually, moving it to `MakeFriends` makes it look not necessary, so it probably isn't. Maybe better a unit test